### ISSUE TITLE
Problem with qutebrowser (and only it) after Mesa upgrade

### ISSUE THREAD

**Issue by intpl** [void-linux/void-packages#34674]
After system upgrade three days ago Qutebrowser started to behave terribly. Entire window become white for milliseconds, scattered parts of window. Pretty much unusable. Chromium and Google Chrome and Firefox which I have installed on the system all work fine. The issue is only with Qutebrowser. It worked perfectly fine before the upgrade.

 I can see in the xbps log these 4 Mesa packages were updated (but also libglapi and libgbm which may also be related):

```
mesa-21.3.2_1
mesa-dri-21.3.2_1
mesa-intel-dri-21.3.2_1
mesa-vulkan-intel-21.3.2_1
```

This is my changed `/etc/modprobe.d/intel-graphics.conf`:
```
options i915 enable_dc=2 enable_fbc=1 fastboot=1
```

I use AwesomeWM if that's related.

**Comment by intpl** [comment]
![qb-scatter](https://user-images.githubusercontent.com/8034626/147201010-6caeff4e-3200-415f-80c4-4f9b33ac9f05.png)
Things like this happening all the time

**Comment by ericonr** [comment]
@ahesford has had the same issues, I believe.

Downgrading the packages locally can fix it temporarily, fwiw.

Possibly https://gitlab.freedesktop.org/mesa/mesa/-/issues/5745

**Comment by ahesford** [comment]
I did have the same issue on Intel Xe and UHD Graphics 630 with new mesa and downgrading the mesa packages resolved it.

**Comment by jsbronder** [comment]
As noted in this bug, exporting `MESA_LOADER_DRIVER_OVERRIDE=i965` appears to have resolved this for me.

https://gitlab.freedesktop.org/mesa/mesa/-/issues/5744

**Comment by intpl** [comment]
> As noted in this bug, exporting `MESA_LOADER_DRIVER_OVERRIDE=i965` appears to have resolved this for me.
> 
> https://gitlab.freedesktop.org/mesa/mesa/-/issues/5744

This fixes it. Thanks! Idk if I should close the issue now

**Comment by ahesford** [comment]
Please leave the issue open. We'll close when the workaround is no longer needed.

**Comment by ahesford** [comment]
I had a chance to try the `MESA_LOADER_DRIVER_OVERRIDE` workaround (it isn't really a fix) on an Intel Xe device, and was met with errors reporting
```
gfx12 and newer are not supported on i965
libGL error: failed to create dri screen
libGL error: failed to load driver: i965
```
There were no glitches, but it seems Mesa fails entirely here. Thus, the workaround seems to be limited to older devices.

It seems that a proper fix may have been merged; maybe we can test a backport pending another upstream release.

**Comment by intpl** [comment]
in case anybody is wondering, installing 21.3.3 from this commit does NOT fix the issue: https://github.com/void-linux/void-packages/pull/34768

**Comment by ardadem** [comment]
[mesa-21.3.3-fix-bug-5731.patch](https://gitlab.freedesktop.org/mesa/mesa/uploads/accd04634117b88139e5320482610013/mesa-21.3.3-fix-bug-5731.patch)

try this patch from https://gitlab.freedesktop.org/mesa/mesa/-/issues/5731

**Comment by ericonr** [comment]
Should be fixed in 1bf20e9d9021baa3bbde781110ab3175f44bcd38

**Comment by ericonr** [comment]
(please reopen if not)
